### PR TITLE
Order and Order Item entities/dto/mapper/repository/service/files were added to application.

### PR DESCRIPTION
**Pull request by oleksandr-chernetskyi**

### CODE REVIEW

File: src/main/java/springboot/onlinebookstore/mapper/order/OrderMapper.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Mappings;
+import springboot.onlinebookstore.config.MapperConfig;
+import springboot.onlinebookstore.dto.order.OrderResponseDto;
+import springboot.onlinebookstore.model.Order;
+
+@Mapper(config = MapperConfig.class)
+public interface OrderMapper {
+    @Mappings({@Mapping(target = "userId", source = "user.id"),
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/mapper/order/OrderMapper.java]: just use @Mapping, without enclosing @mappings

File: src/main/java/springboot/onlinebookstore/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted = false")
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/model/Order.java]:
```suggestion
@SQLDelete(sql = "UPDATE orders SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(clause = "is_deleted = FALSE")
```

File: src/main/java/springboot/onlinebookstore/model/OrderItem.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE order_items SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted = false")
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/model/OrderItem.java]: same

File: src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    public OrderResponseDto createOrder(Long id, OrderRequestDto orderRequestDto) {
+        User authenticatedUser = userService.getAuthenticatedUser();
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .getUserById(authenticatedUser.getId())
+                .orElseThrow(() -> {
+                    log.error("Create order method failed. "
+                            + "Can't find shopping cart by user ID: {}",
+                            authenticatedUser.getId());
+                    throw new NoShoppingCartException("Shopping cart not found by user");
+                });
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java]: extract to a separate method, this one is bloated

File: src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @Override
+    public OrderItemResponseDto findOrderItemById(Long orderId, Long itemId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> {
+                    log.error("Find order item method failed. "
+                                    + "Can't find order by ID: {}",
+                            orderId);
+                    throw new EntityNotFoundException("Can't find order by existing ID");
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java]: always include id in such exception messages

File: src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        List<Order> orders = orderRepository.findAllOrdersByUserId(userId);
+        return orders.stream()
+                .map(order -> {
+                    OrderResponseDto orderResponseDto = orderMapper.toDto(order);
+                    Set<OrderItemResponseDto> orderItemResponseDtos
+                            = order.getOrderItems().stream()
+                            .map(orderItemMapper::toDto)
+                            .collect(Collectors.toSet());
+                    orderResponseDto.setOrderItems(orderItemResponseDtos);
+                    return orderResponseDto;
```
Comment (okuzan) [inline src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java]: to a private method

File: src/main/resources/db/changelog/changes/22-insert-order-items-into-order-items-table.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+                  value: "2"
+              - column:
+                  name: "book_id"
+                  value: "2"
+              - column:
+                  name: "quantity"
+                  value: "2"
+              - column:
+                  name: "price"
+                  value: "30"
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/22-insert-order-items-into-order-items-table.yaml]:
```suggestion
                  value: "30"

```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by koval-kn — APPROVED** [review]

File: src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+                            authenticatedUser.getId());
+                    throw new NoShoppingCartException("Shopping cart not found by user");
+                });
+    }
+
+    private Order createOrderFromCart(ShoppingCart shoppingCart,
+            OrderRequestDto orderRequestDto) {
+        Order order = new Order();
+        order.setUser(shoppingCart.getUser());
+        order.setTotal(BigDecimal.ZERO);
```
Comment (Rommelua) [inline src/main/java/springboot/onlinebookstore/service/impl/OrderServiceImpl.java]:
```suggestion
```

File: src/main/java/springboot/onlinebookstore/validation/status/Status.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Constraint(validatedBy = StatusValidator.class)
+@Target({ElementType.TYPE, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Status {
```
Comment (Rommelua) [inline src/main/java/springboot/onlinebookstore/validation/status/Status.java]: delete this annotation.
We should not create unique annotation for each enum in the app

**Pull request merged**